sdks/python/apache_beam/runners/common.py

```diff
@@ -842,29 +847,37 @@ def _invoke_process_per_window(self,
 
   def try_split(self, fraction):
     # type: (...) -> Optional[Tuple[SplitResultPrimary, SplitResultResidual]]
-    if self.threadsafe_restriction_tracker and self.current_windowed_value:
+    if not self.is_splittable:
+      return None
+
+    with self.splitting_lock:
```

Comment:
If we are deep copying objects, it seems like we can use the lock to guard the copying logic only, instead of the entire split logic.

Reply:
Not deep copying objects.